src/Sylius/Bundle/ApiBundle/StateProvider/ShopUserResetPasswordProvider.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProvider;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\Patch;
+use ApiPlatform\State\ProviderInterface;
+use Sylius\Bundle\ApiBundle\Command\Account\ResetPassword;
+use Webmozart\Assert\Assert;
+
+/**
+ * @experimental
+ *
+ * @implements ProviderInterface<ResetPassword>
+ */
+final readonly class ShopUserResetPasswordProvider implements ProviderInterface
+{
+    /**
+     * @throws \Exception
+     */
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): ResetPassword
+    {
+        Assert::same($operation->getClass(), ResetPassword::class);
+
+        if ($operation instanceof Patch) {
```

Comment:
The `provider` is set on resource `Sylius\Bundle\ApiBundle\Command\Account\ResetPassword`, and there are 2 operations: `POST` and `PATCH`. So why is only `PATCH` here?

Reply:
This is a good observation. I overlooked this during the implementation.
I did some investigation and it looks like the custom `Provider` with the `POST` operation is skipped because of this condition:
https://github.com/api-platform/core/blob/main/src/Symfony/EventListener/ReadListener.php#L71-L73

I'm not sure if this is a good conclusion - we can't use ``Provider`` with ``POST`` operations 🤔.

Reply:
Therefore, I'm not sure if I should apply any changes to this class, because for now it doesn't mess up the system